Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.Data.cs

```diff
@@ -33,6 +41,55 @@ public UvMeasure(Orientation orientation, double width, double height)
                     V = width;
                 }
             }
+
+            public UvMeasure Add(double u, double v)
+                => new UvMeasure { U = U + u, V = V + v };
+        }
+
+        private struct UvRect
+        {
+            public UvMeasure Position { get; set; }
+
+            public UvMeasure Size { get; set; }
+
+            public Rect ToRect(Orientation orientation)
+            {
+                switch (orientation)
+                {
+                    case Orientation.Vertical:
+                        return new Rect(Position.V, Position.U, Size.V, Size.U);
+                    case Orientation.Horizontal:
+                        return new Rect(Position.U, Position.V, Size.U, Size.V);
+                    default:
+                        throw new NotSupportedException();
+                }
+            }
+        }
+
+        private class Row
+        {
+            private readonly List<UvRect> _childrenRects;
+            private UvMeasure _rowSize;
+
+            public Row()
+            {
+                _childrenRects = new List<UvRect>();
+                _rowSize = UvMeasure.Zero;
+            }
+
+            public IReadOnlyList<UvRect> ChildrenRects => _childrenRects;
+
+            /// <summary>
+            /// Gets the size of the row.
+            /// </summary>
+            public UvMeasure Size => _rowSize;
+
+            public void Add(UvMeasure position, UvMeasure size)
+            {
+                _childrenRects.Add(new UvRect { Position = position, Size = size });
+                _rowSize.U = Math.Max(_rowSize.U, position.U + size.U);
```

Comment:
`position.U + size.U` should always be larger than _rowSize.U. No need for the extra check